### DISCUSSION TITLE
NullPointerException for enum type

### DISCUSSION THREAD

**Discussion by Perry45** [sngular/kloadgen#277] [Q&A]
Hello,

in general my setup works and a message is sent from JMeter / KLoadGen to my application.

My only problem is, that I have to set a enum to null. If I try set it properly I get this error:
Caused by: java.lang.NullPointerException: null of de.mycomp.avro.model.FruitType of union in field fruitType of
de.mycomp.avro.model.Fruit of array of union in field fruits of
de.mycomps.avro.model.FruitList of union in field fruitList of
de.mycomp.avro.model.AggregateAttribute in field aggregateAttribute of
de.mycomp.avro.model.MyFruitsValue

The Schema looks like:
`{
  "type": "record",
  "name": "MyFruitsValue",
  "namespace": "de.mycomp.avro.model",
  "fields": [
    {
      "name": "aggregateAttribute",
      "type": {
        "type": "record",
        "name": "AggregateAttribute",
        "fields": [
          {
            "name": "fruitList",
            "type": [
              "null",
              {
                "type": "record",
                "name": "FruitList",
                "fields": [
                  {
                    "name": "fruits",
                    "type": [
                      "null",
                      {
                        "type": "array",
                        "items": {
                          "type": "record",
                          "name": "Fruit",
                          "fields": [
                            {
                              "name": "fruitType",
                              "type": [
                                "null",
                                {
                                  "type": "enum",
                                  "name": "FruitType",
                                  "symbols": [
                                    "MY_ENUM_1",
                                    "MY_ENUM_2",
                                    "MY_ENUM_3",
                                  ]
                                }
                              ],
                              "default": null
                            },
                          ]
                        }
                      }
                    ],
                    "default": null
                  }
                ]
              }
            ],
            "default": null
          },
        ]
      },
    }
  ]
}`

I try to send it like so:
aggregateAttribute.fruitList.fruits[].fruitType enum 16 [MY_ENUM_1]

The other values (which are strings) in the array fruits[] are working and received by my app properly.

I assume it has something to do with `                              "type": [
                                "null",
                                {
                                  "type": "enum",
                                  "name": "FruitType",
                                  "symbols": [
                                    "MY_ENUM_1",
                                    "MY_ENUM_2",
                                    "MY_ENUM_3",
                                  ]
                                }` because of the null or default null?!

**Comment by jemacineiras** [comment]
Hi @Perry45,
we can take a look into it. Do you mind open us an issue?

Cheers

**Reply by Perry45** [reply]
Would be great. I opened one.
